flink-streaming-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumer.java

```diff
@@ -563,37 +559,44 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
 					LOG.warn("Received confirmation for unknown checkpoint id {}", checkpointId);
 					return;
 				}
-	
-				checkpointOffsets = (long[]) pendingCheckpoints.remove(posInMap);
+
+				//noinspection unchecked
+				checkpointOffsets = (HashMap<KafkaTopicPartition, Long>) pendingCheckpoints.remove(posInMap);
+
 				
 				// remove older checkpoints in map
 				for (int i = 0; i < posInMap; i++) {
 					pendingCheckpoints.remove(0);
 				}
 			}
-	
-			if (LOG.isInfoEnabled()) {
-				LOG.info("Committing offsets {} to offset store: {}", Arrays.toString(checkpointOffsets), offsetStore);
+			if(checkpointOffsets.size() == 0) {
```

Comment:
I think this is a bug. If the list is empty after you remove the checkpoint to commit, it only means that no pending checkpoint was subsumed. It should still commit the offsets to Kafka.